### ISSUE TITLE
[Bug]:  conversion failed from VARCHAR to INTEGER

### ISSUE THREAD

**Issue by Dream95** [matrixorigin/matrixone#4926]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):v0.5.1
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

#WITHOUT CAST FUNCTION
mysql> SELECT 'a' = 1;
ERROR 1105 (HY000): Can't cast 'a' from VARCHAR type to BIGINT type.
#WITH CAST FUNCTION
mysql> SELECT cast( 'a' as INTEGER);
ERROR 1105 (HY000): Can't cast 'a' from VARCHAR type to INT type.
mysql> SELECT cast('a' as UNSIGNED);
ERROR 1105 (HY000): Can't cast 'a' from VARCHAR type to BIGINT UNSIGNED type.

### Expected Behavior

conversion success

**Comment by jianwan0214** [comment]
the string 'a' can not cast to bigint type, is it not right? if you run select '1'=1; you will get true.

**Comment by jianwan0214** [comment]
this is meets expectations, you can contact zhangxiao.

**Comment by aressu1985** [comment]
@domingozhang pls confirm this 

now mo is :
mysql> SELECT 'a' = 1;
ERROR 20203 (HY000): invalid argument cast to int, bad value a
mysql> SELECT cast( 'a' as INTEGER);
ERROR 20203 (HY000): invalid argument cast to int, bad value a
mysql> SELECT cast('a' as UNSIGNED);
ERROR 20203 (HY000): invalid argument cast to uint, bad value a
mysql>

**Comment by domingozhang** [comment]
MO has different rule of conversation rules. These actions are normal.

**Comment by aressu1985** [comment]
fixed:e23e00f1d01d43e33c509a6d239ccc38b5f93409

mysql> SELECT 'a' = 1;
ERROR 20203 (HY000): invalid argument cast to int, bad value a
mysql> SELECT cast( 'a' as INTEGER);
ERROR 20203 (HY000): invalid argument cast to int, bad value a
mysql> SELECT cast('a' as UNSIGNED);
ERROR 20203 (HY000): invalid argument cast to uint, bad value a
mysql>
